.teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import common.BuildToolBuildJvm
+import common.Os
+import common.VersionedSettingsBranch
+import common.gradleWrapper
+import common.javaHome
+import common.requiresOs
+import common.toCapitalized
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import vcsroots.gradleMasterVersionedSettings
+import vcsroots.useAbsoluteVcs
+
+object PublishKotlinDslPlugin : BuildType({
+    name = "Publish Kotlin DSL Plugin"
+    id("Util_PublishKotlinDslPlugin")
+    uuid = "${VersionedSettingsBranch.fromDslContext().branchName.toCapitalized()}_Util_PublishKotlinDslPlugin"
+    vcs.useAbsoluteVcs(gradleMasterVersionedSettings)
+
+    requirements {
+        requiresOs(Os.LINUX)
+    }
+
+    params {
+        param("env.JAVA_HOME", javaHome(BuildToolBuildJvm, Os.LINUX))
+        param("env.GRADLE_PUBLISH_KEY", "%plugin.portal.publish.key%")
+        param("env.GRADLE_PUBLISH_SECRET", "%plugin.portal.publish.secret%")
```

Comment:
Should this be marked as secret somehow?

Reply:
`plugin.portal.publish.key` is already defined in Root project so I thought we can reference it directly...